src/sentry/notifications/helpers.py

```diff
@@ -610,3 +641,149 @@ def get_providers_for_recipient(
     user_providers = [get_provider_enum_from_string(idp_type) for idp_type in idp_types]
     user_providers.append(ExternalProviders.EMAIL)  # always add in email as an option
     return user_providers
+
+
+def recipient_is_user(recipient: RpcActor | Team | User) -> bool:
+    from sentry.interfaces.user import User
+
+    if isinstance(recipient, RpcActor) and recipient.actor_type == ActorType.USER:
+        return True
+    return isinstance(recipient, (RpcUser, User))
+
+
+def recipient_is_team(recipient: RpcActor | Team | User) -> bool:
+    from sentry.models.team import Team
+
+    if isinstance(recipient, RpcActor) and recipient.actor_type == ActorType.TEAM:
+        return True
+    return isinstance(recipient, Team)
+
+
+def get_all_setting_providers(
+    recipient: RpcActor | Team | User,
+    project: Project | None = None,
+    organization: Organization | None = None,
+) -> Iterable[NotificationSettingProvider]:
+    project_settings = (
+        Q(scope_type=NotificationScopeType.PROJECT.value, scope_identifier=project.id)
```

Comment:
@snigdhas For these project and org queries, we need to pass in either the `user_id` or the `team_id`. Someone else's org-scoped settings shouldn't apply to you.